comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -38,8 +41,10 @@ const auditFileBasename = "secret-audit-file.json"
 type provides struct {
 	fx.Out
 
-	Comp          secrets.Component
-	FlareProvider flaretypes.Provider
+	Comp            secrets.Component
+	FlareProvider   flaretypes.Provider
+	InfoEndpoint    api.AgentEndpointProvider
+	RefreshEndpoint api.AgentEndpointProvider
```

Comment:
Have to tested the the flatten of value groups? It might allow us avoiding having to create a separate field per API route https://pkg.go.dev/go.uber.org/fx#hdr-Value_group_flattening

Reply:
I did and it does not work as we hoped, the `flatten` annotation needs to be done on the provider and *not* the consumer

the point of using the `api.AgentEndpointProvider` was so users did not need to add the group tags themselves in every component, to use it we'd need to instead have each component do that themselves and. not use the constructor

```
type provides struct {
    fx.Out

    Comp            secrets.Component
    FlareProvider   flaretypes.Provider
    Endpoints       []api.EndpointProvider  `group:"agent_endpoint,flatten"`
}
```

Reply:
maybe we could have another constructor for slices to register multiple endpoints?